server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -4013,6 +4013,8 @@ public void afterRefresh(boolean didRefresh) {
             if (enableFieldHasValue) {
                 try (Engine.Searcher hasValueSearcher = getEngine().acquireSearcher("field_has_value")) {
                     setFieldInfos(FieldInfos.getMergedFieldInfos(hasValueSearcher.getIndexReader()));
+                } catch (AlreadyClosedException ignore) {
+                    // engine is closed - no updated FieldInfos is fine
```

Comment:
should we add a specific test that verifies the behaviour when the engine is closed?